src/Serializer/AbstractItemNormalizer.php

```diff
@@ -1163,9 +1163,10 @@ private function createAndValidateAttributeValue(string $attribute, mixed $value
             }
 
             if (
-                ($t instanceof CollectionType && $collectionValueType instanceof ObjectType && (null !== ($className = $collectionValueType->getClassName())))
-                || ($t instanceof LegacyType && $t->isCollection() && null !== $collectionValueType && null !== ($className = $collectionValueType->getClassName()))
+                ($t instanceof CollectionType && $collectionValueType instanceof ObjectType)
```

Comment:
In this condition `(null !== ($className = $collectionValueType->getClassName()))` is always true cause `getClassName` returns a string.

So we need to remove it from the condition, and it force declaring the variable after the if.